main.tf

```diff
@@ -158,23 +216,56 @@ resource "aws_security_group" "this" {
 }
 
 resource "aws_security_group_rule" "ingress" {
-  count                    = var.create_security_group ? 1 : 0
-  type                     = var.ingress_type
+  for_each                 = var.ingress_rules
+  type                     = "ingress"
   description              = "Allow inbound traffic from existing Security Groups"
-  from_port                = var.port
-  to_port                  = var.port
-  protocol                 = var.ingress_protocol
+  from_port                = lookup(each.value, "from_port")
+  to_port                  = lookup(each.value, "to_port")
+  protocol                 = "tcp"
   source_security_group_id = join("", aws_security_group.this.*.id)
```

Comment:
Hey @JohnNMwangi , I think it would be good thing to let `source_security_group_id` argument be optional. The rest are good